src/EventLoop/FiberLocal.php

```diff
@@ -39,31 +36,38 @@ private static function getFiber(): \Fiber
             });
         }
 
-        return $fiber;
+        $localStorage = self::$localStorage ??= new \WeakMap();
+        return $localStorage[$fiber] ??= new \WeakMap();
     }
 
-    public function __construct(mixed $value)
+    /**
+     * @param \Closure():T $initializer
+     */
+    public function __construct(private \Closure $initializer)
     {
-        $this->set($value);
     }
 
+    /**
+     * @param T $value
+     *
+     * @return void
+     */
     public function set(mixed $value): void
     {
-        $fiber = self::getFiber();
-        $localStorage = self::getLocalStorage();
-
-        if (!isset($localStorage[$fiber])) {
-            $localStorage[$fiber] = new \WeakMap();
-        }
-
-        $localStorage[$fiber][$this] = $value;
+        self::getFiberStorage()[$this] = $value;
     }
 
+    /**
+     * @return T
+     */
     public function get(): mixed
     {
-        $fiber = self::getFiber();
-        $localStorage = self::getLocalStorage();
+        $fiberStorage = self::getFiberStorage();
+
+        if (!isset($fiberStorage[$this])) {
```

Comment:
Do we want to allow `null` as being initialized?

Reply:
Ah yes, that was my intension. I guess this needs to be `$fiberStorage->offsetExists($this)`?

Reply:
I hope so, yes.